h2/events.py

```diff
@@ -343,15 +343,16 @@ def from_settings(cls, old_settings, new_settings):
         """
         e = cls()
         for setting, new_value in new_settings.items():
+            setting = _setting_code_from_int(setting)
             original_value = old_settings.get(setting)
             change = ChangedSetting(setting, original_value, new_value)
             e.changed_settings[setting] = change
 
         return e
 
     def __repr__(self):
-        return "<RemoteSettingsChanged changed_settings:%s>" % (
-            self.changed_settings,
+        return "<RemoteSettingsChanged changed_settings:{%s}>" % (
+            ", ".join([repr(cs) for cs in self.changed_settings.values()]),
```

Comment:
Small note here, but you don't need the list. `join` works just fine on generators, so we can save ourselves the allocation of an intermediate list here and below.